dotnet/src/Agents/OpenAI/Extensions/ModelConfigurationExtensions.cs

```diff
@@ -16,15 +16,13 @@ internal static class ModelConfigurationExtensions
     /// Gets the endpoint property as a <see cref="Uri"/> from the specified <see cref="ModelConfiguration"/>.
     /// </summary>
     /// <param name="configuration">Model configuration</param>
-    internal static Uri GetEndpointUri(this ModelConfiguration configuration)
+    internal static Uri? GetEndpointUri(this ModelConfiguration configuration)
```

Comment:
Since is internal, we can change to `Try`. I don't think we need to add `Uri` to the name of the method as the return type is descriptive.

```suggestion
    internal static Uri? TryGetEndpoint(this ModelConfiguration configuration)
```